### PR TITLE
[CL]: Fix low level tick rounding to mitigate fund drain bug

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #5516 

## What is the purpose of the change

This PR implements the discussed fix to the vector outlined in #5493. It currently breaks tick invariant tests, which are skipped in this PR until we converge on how to handle tick <> sqrt price conversions in #5519.

This PR also renames `PriceToTickRoundDown` to `PriceToTickRoundDownSpacing`, as the original naming caused some confusion around how this bug could be possible in the first place.

## Testing and Verifying

- Original attack vector outlined in #5493 is converted to an edge case test and no longer drains funds
- A number of faulty test vectors are amended (numbers were inadvertently rounded up, leading to incorrect expected values that should now be fixed)
- All other existing tests pass without issue
- **Important:** tick to price invariant tests are now skipped as they no longer hold due to sqrt precision loss. This does not seem to cause any immediate issues, but needs to be kept in mind if we agree not to pursue #5519.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/concentrated-liquidity/swaps_test.go [hunk 1]
```diff
@@ -411,14 +411,14 @@ var (
 			spreadFactor:  sdk.ZeroDec(),
 			// params
 			// liquidity: 		 1517882343.751510418088349649
-			// sqrtPriceNext:    70.668238976219012613 which is 4994 https://www.wolframalpha.com/input?i=70.710678118654752440+%2B+42000000+%2F+1517882343.751510418088349649
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/swaps_test.go:414]: Note to reviewer: this test vector seemed to have been incorrectly set up (previous wolfram link went to an unrelated equation)

File: x/concentrated-liquidity/math/tick.go [hunk 2]
```diff
@@ -108,18 +108,21 @@ func PriceToTick(price sdk.Dec) (int64, error) {
 
 	// Determine the tick that corresponds to the price
 	// This does not take into account the tickSpacing
-	tickIndex := CalculatePriceToTick(price)
+	tickIndex, err := CalculatePriceToTick(price)
+	if err != nil {
+		return 0, err
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/math/tick.go:113]: Note to reviewer: returning 0 here isn't ideal, but it seems to be what we do in all other tick logic upon error. If there's a better value to return here that's more clearly invalid (e.g. an `sdk.Int{}` equivalent) please lmk

File: x/concentrated-liquidity/math/tick.go [hunk 3]
```diff
@@ -206,7 +210,64 @@ func CalculatePriceToTick(price sdk.Dec) (tickIndex int64) {
...
+		// If price is lower than the price in the tick below, our error assumption is violated so we panic.
+		if price.LT(tickBelowPrice) {
+			panic(fmt.Sprintf("price %s is outside of error bounds for tick %d", price, tickIndex))
+		}
+
+		// If price falls in the bucket below `tickIndex`, return that bucket index.
+		// Otherwise, return the current tick index.
+		if price.GTE(tickBelowPrice) && price.LT(curPrice) {
+			return tickIndex - 1, nil
+		}
```
Comment (ValarDragon) [inline x/concentrated-liquidity/math/tick.go:243]: I think we should make a new SqrtPriceToTick function, that does the squaring internally, then does this comparison on the derived sqrt prices. As the problem is our price definition doesn't match due to errors.

(I don't think this on its own will save us from anything aside from the erroneous bucket definition due to not truncating)

**Comment by AlpinYukseloglu** [comment]
Closing in favor of #5541

**Pull request closed without merging**